python/CLOUDTRAIL_S3_DATAEVENTS_ENABLED/CLOUDTRAIL_S3_DATAEVENTS_ENABLED.py

```diff
@@ -0,0 +1,489 @@
+  Scenario: 3
+    Given: No AWS CloudTrail trail exists
+     Then: Return NON_COMPLIANT
+  Scenario: 4
+    Given: At least one AWS CloudTrail trail exists
+      And: None of the trails 'HasCustomEventSelectors' set as 'True'
+     Then: Return NON_COMPLIANT
+  Scenario: 5
+    Given: At least one AWS CloudTrail trail exists
+      And: At least one of the trails 'HasCustomEventSelectors' set as 'True'
+      And: The Type in 'DataResources' returned from 'get_event_selectors' is not AWS::S3::Object
+     Then: Return NON_COMPLIANT
+  Scenario: 6
+    Given: Rule parameter is provided and valid
+      And: At least one AWS CloudTrail trail exists
+      And: At least one of the trails 'HasCustomEventSelectors' set as 'True'
+      And: The Type in 'DataResources' returned from 'get_event_selectors' is AWS::S3::Object
+      And: The Value in 'DataResources' returned from 'get_event_selectors' does not include at least one S3 bucket as specified in rule parameter 'S3BucketName'.
+     Then: Return NON_COMPLIANT
+  Scenario: 7
+    Given: Rule parameter is provided and valid
+      And: At least one AWS CloudTrail trail exists
+      And: At least one of the trails 'HasCustomEventSelectors' set as 'True'
+      And: The Type in 'DataResources' returned from 'get_event_selectors' is AWS::S3::Object
+      And: The Value in 'DataResources' returned from 'get_event_selectors' includes the S3 bucket as specified in rule parameter 'S3BucketName'.
+     Then: Return COMPLIANT
+  Scenario: 8
+    Given: Rule parameter value is empty
+      And: At least one AWS CloudTrail trail exists
+      And: At least one of the trails 'HasCustomEventSelectors' set as 'True'
+      And: The Type in 'DataResources' returned from 'get_event_selectors' is AWS::S3::Object
+      And: The Value in 'DataResources' returned from 'get_event_selectors' is arn:aws:s3:::
+     Then: Return COMPLIANT
+'''
+import json
+import sys
+import datetime
+import re
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+DEFAULT_CHECK = 'arn:aws:s3'
+
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    ct_client = get_client('cloudtrail', event)
+    evaluations = []
+    compliant_buckets = []
+    non_compliant_buckets = []
+    trail_list = get_all_trails(ct_client)
+
+    #No trails configured: return NON_COMPLIANT
+    if not trail_list:
+        evaluations.append(build_evaluation(event['accountId'], 'NON_COMPLIANT', event, DEFAULT_RESOURCE_TYPE, 'There are no AWS CloudTrail Trails in the region.'))

```

Comment:
We need to change the annotation here: 'There are no AWS CloudTrail Trails configured to log events in this region.'